### PR TITLE
Add Lua grammar

### PR DESCRIPTION
**Pull request by Ruin0x11**
Part of https://github.com/returntocorp/semgrep/issues/1931.

### CODE REVIEW

**Comment by aryx** [comment]
I didn't even get notified of this PR. Were you able to request a reviewer @Ruin0x11 ?

**Comment by aryx** [comment]
Or maybe just owner of the github project can do that?

File: src/semgrep-lua/grammar.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/*
+ * semgrep-kotlin
```
Comment (aryx) [inline src/semgrep-lua/grammar.js:2]: I guess semgrep-lua :)

**Review by aryx — APPROVED** [review]
not sure why this PR also include changes to tree-sitter-ruby.

**Review by aryx — CHANGES_REQUESTED** [review]
Could you remove the change to tree-sitter-ruby and revert to the old version of tree-sitter-ruby submodule? There is no reason this PR should update also tree-sitter-ruby.
We should just add tree-sitter-lua.

**Comment by aryx** [comment]
Moreover your changes to tree-sitter-ruby seems to go to a previous version, so we would regress.

**Comment by aryx** [comment]
ping

**Comment by mjambon** [comment]
Thank you @Ruin0x11 ! It looks great. We're very impressed that you were able to deal with all these repositories to add lua support.

I have the same remarks as @aryx . The tree-sitter-ruby submodule should be at 694f81c (Max Brunsfeld, Dec 8 2020).

**Review by mjambon — CHANGES_REQUESTED** [review]
See the changes requested by @aryx .

**Comment by mjambon** [comment]
Next we'll have to extend the tree-sitter-lua grammar with semgrep constructs `...` and metavariables (`$FOO`) for it to parse semgrep/lua patterns in addition to pure lua programs. I can help with that, as I'm going through this process for ruby this week.

**Comment by aryx** [comment]
You mean C#?

**Comment by mjambon** [comment]
@aryx wrote:
> You mean C#?
yes

**Comment by aryx** [comment]
ok, I'll try tomorrow to fix this PR if @Ruin0x11 does not. Then I'll try to extend the grammar to allow metavariables! so we can start to write semgrep Lua patterns!

**Comment by Ruin0x11** [comment]
@aryx Sorry, I got caught up in life for a moment. Should be fixed now.

**Review by aryx — APPROVED** [review]

**Comment by aryx** [comment]
Sure, no problem @Ruin0x11 , you've done a lot already.

**Pull request merged**